packages/trader/src/Modules/Trading/Components/Form/TradeParams/Duration/duration-mobile.jsx

```diff
@@ -167,8 +167,10 @@ const Numbers = observer(
         };
 
         const setExpiryDate = (epoch, duration) => {
+            if (trade_duration_unit !== 'd') {
+                return moment.utc().add(Number(duration), 'days').format('D MMM  YYYY, [23]:[59]:[59] [GMT +0]');
```

Comment:
@kate-deriv it looks like there are 2 spaces between `MMM`  and `YYYY`, please check. Please also elaborate on the reasons behind trying to avoid sending a new `proposal` request with updated `duration_unit` every time a user switches to another Duration tab on mobile, whereas on desktop we are sending this new proposal request every time it happens to get fresh data from `proposal`. Are there any reports of any negative impact these requests are causing?🙏🏻

Reply:
@maryia-deriv thank you for noticing, it's extra space) I'll remove it)))
About approach: I chose between several options. First one was to send request to BE when user switched the duration tab to 'days'. In that case FE will receive correct data of expiry, but the default chosen value will also change to days even if user didn't apply it (by clicking on 'Ok' button). Another drawbacks are: 1) we will send to much requests to BE 2) sending request -> getting answer from BE -> processing data on FE -> rendering takes about 1.5 sec (at least on my IOS when I tested that approach), so user will se wrong expiry date first and only after 1.5 sec correct one (on Desktop in production we have same problem).
Another way was to hardcode the expiry date if  'minutes'/'hours' were chosen by default. The reason why it was possible to hardcode is that time of expiry is always the same (at 23:59:59 GMT +0). So I only need to calculate the day taking in mind extra duration which can add user. The drawback is obvious - I hardcoded time (at least not all date). The benefit is immediate showing correct Expiry date to user without sending extra request and any delays. 
Before taking the final decision I ask FE members to stay after daily call and discuss pros and cons. So the solution with hardcoding was considered to be the better option